Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs

```diff
@@ -164,6 +204,38 @@ private PropertyViewModel GetViewModel (IPropertyInfo property)
 			return new StringPropertyViewModel (property, this.editors);
 		}
 
+		void FilterData ()
+		{
+			this.properties.Clear ();
+
+			UpdateProperties ();
+		}
+
+		List<PropertyViewModel> cachedProperties;
+		public List<PropertyViewModel> CachedProperties {
+			get {
+				if (cachedProperties == null) {
+					cachedProperties = Properties.ToList ();
+				}
+				return cachedProperties;
+			}
+		}
+
+		IEnumerable<IGrouping<string, PropertyViewModel>> cachedPropertiesGroupBy;
+		public IEnumerable<IGrouping<string, PropertyViewModel>> CachedPropertiesGroupBy {
```

Comment:
The consuming UI code doesn't need to know this is cached, no need to leak implementation details into the API naming.